src/engine/controls/bpmcontrol.cpp

```diff
@@ -676,7 +687,10 @@ double BpmControl::getNearestPositionInPhase(
     } else if (this_near_next && !other_near_next) {
         dNewPlaypos += dThisNextBeat;
     } else { //!this_near_next && other_near_next
-        dThisPrevBeat = pBeats->findNthBeat(dThisPosition, -2);
+        dThisPrevBeat =
+                m_pBeats->findNthBeat(mixxx::Frame(dThisPosition / 2.0), -1)
```

Comment:
These recurring conversions are difficult to identify. Please use inlined utility functions for this purpose, e.g. samples2frames()/frames2samples()